nav2_lifecycle_manager/src/lifecycle_manager.cpp

```diff
@@ -78,19 +75,23 @@ LifecycleManager::LifecycleManager()
   transition_label_map_[Transition::TRANSITION_DEACTIVATE] = std::string("Deactivating ");
   transition_label_map_[Transition::TRANSITION_UNCONFIGURED_SHUTDOWN] =
     std::string("Shutting down ");
+}
 
+LifecycleManager::~LifecycleManager()
+{
+  RCLCPP_INFO(get_logger(), "Destroying %s", get_name());
+}
+
+void
+LifecycleManager::init()
```

Comment:
So I understand why you did this, you wanted to use `shared_from_this()` which can't be called from the constructor, but we need to find another way. Creating the `init` function that the `main` must call is not a sufficient solution because then it breaks the ability of this server to ever become a ROS2 component node. 

Another options might be to create a timer on a 100ms timeout created at the end of the constructor that triggers this initialization behavior in a lambda.